### PR TITLE
feat(glucose): add quadratic log loss

### PR DESCRIPTION
**Pull request by scaubrey**
Adding quadratic log loss glucose metric

### CODE REVIEW

File: tidepool_data_science_metrics/glucose/glucose.py [hunk 1]
```diff
@@ -376,6 +376,31 @@ def lbgi_risk_score(lbgi: np.float64) -> int:
...
+
+    gc : float, optional
+        The set point, which is the minimum of the quadratic
+
+    Returns
+    -------
+    float: loss
+    """
+    _validate_bg(bg_array)
+    loss = np.mean(np.power(np.log(bg_array) - np.log(gc), 2))
```
Comment (ed-nykaza) [inline tidepool_data_science_metrics/glucose/glucose.py:400]: I think the formation you have is fine, but I think we could simplify by combining the logs --> np.mean(np.power(np.log(bg_array/gc), 2))

**Review by ed-nykaza — APPROVED** [review]
LGTM

File: tidepool_data_science_metrics/glucose/glucose.py (hunk 1 above)
Comment (scaubrey) [reply tidepool_data_science_metrics/glucose/glucose.py:400]: Yep, that's a good option. I might prefer the current form for clarity, but I could be convinced either way. Your suggestion is slightly faster as shown with a quick test below, so we should change it if we find ourselves in need of performance improvements with the metrics. Notably, computing the loss with the scipy geo mean/std is slower than both, but I think the big benefit of the geo mean/std version is easy conversion when you already have the arithmetic mean/std.

```In [71]: %%timeit
    ...: for i in range(int(1e4)):
    ...:     bg_array = np.random.random(1000)
    ...:     geo_mean = gmean(bg_array)
    ...:     geo_std = gstd(bg_array)
    ...:     np.power(np.log(geo_std), 2) + np.power(np.log(geo_mean) - np.log(gc), 2)
    ...:
2.16 s ± 48 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [72]: %%timeit
    ...: for i in range(int(1e4)):
    ...:     bg_array = np.random.random(1000)
    ...:     np.mean(np.power(np.log(bg_array) - np.log(gc), 2))
    ...:
1.46 s ± 51.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [73]: %%timeit
    ...: for i in range(int(1e4)):
    ...:     bg_array = np.random.random(1000)
    ...:     np.mean(np.power(np.log(bg_array/gc), 2))
    ...:
1.35 s ± 54.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)```
